ADR/user-specified-tokens-build.md

```diff
@@ -0,0 +1,28 @@
+# Enable users to configure tokens/credentials for a Build/Test Pipeline
+
+Date: 2023-03-03
+
+## Status
+
+Provisional
+
+
+## Context
+
+As a user, I would like to configure sensitive information/tokens/credentials for the build/test pipeline to consume as part of a partner task.
+
+
+## Decision
+
+1. UI will take in sensitive information as form input and upload it to SPI using the user's authentication token. See https://github.com/redhat-appstudio/service-provider-integration-operator/blob/main/docs/USER.md#uploading-access-token-to-spi-using-kubernetes-secret .
+2. In the above form, the user should be able to choose the name of the `Secret` from a dropdown. 
+The dropdown would be a pre-populated list of `Secret` names associated with known partner integrations. The user should also be allowed to provide a name barring the ones that already exist in the namespace.
+3. The UI will create an `SPIAccessTokenBinding` CR that would create and link the secret with the _pipeline_ `serviceaccount`. See https://github.com/redhat-appstudio/service-provider-integration-operator/blob/main/docs/USER.md#linking-a-secret-to-a-pre-existing-service-account .
```

Comment:
I guess that you are referring to the pipeline service account in the user's namespace?
If so, when a user has multiple environments (which in the first implementation map to namespaces), in which namespace the integration pipeline runs?

Reply:
The scope of the secrets should definitely be mentioned in the document -- i.e. whether they are only in the target workspace or more broadly set.

Reply:
Yes, in the user's namespace at the moment. I checked with @dirgim , he mentioned that, at the moment, we are going to be running integration tests in the user's namespace itself.

I thought it would be implicit but SPI today supports SPIAccessTokenBinding in the same namespace only. Let me update the ADR.